agent/hcp/manager.go

```diff
@@ -116,6 +118,12 @@ func (m *Manager) Run(ctx context.Context) {
 			m.logger.Error("failed to send server status to HCP", "err", err, "next_heartbeat", nextUpdate.String())
 		}
 
+		// Check for configured management token from HCP. It MUST NOT override the user-provided initial management token.
+		if hcpManagement := m.cfg.CloudConfig.ManagementToken; len(hcpManagement) > 0 {
```

Comment:
I attempted to run this code before the "main loop" of this manager, but it errored with the following log because the node did not become the leader yet:
```
2024-01-02T15:50:46.669-0500 [ERROR] agent.hcp_manager: failed to initialize HCP management token: err="failed to bootstrap with (new) HCP Management Token: node is not the leader"
```

While there might be a way to ensure this happens after the node becomes the leader, it feels a bit more robust to have this code running in the main loop — this way, we could potentially self-heal from any timing (or other) issues.

Reply:
This should be safe as long as the passed in `InitializeManagementTokenFn` is idempotent. The one that is currently used from `server.go` is idempotent as it is a no-op if the specified token already exists.